CollapseLauncher/Classes/Helper/Database/DBHandler.cs

```diff
@@ -0,0 +1,304 @@
+        #region Config Properties
+
+        private static bool? _enabled;
+        public static bool IsEnabled
+        {
+            get
+            {
+                if (_enabled != null) return (bool)_enabled;
+                var c = DbConfig.DbEnabled;
+                _enabled = c;
+                return c;
+            }
+            set
+            {
+                _enabled           = value;
+                DbConfig.DbEnabled = value;
+
+                _isFirstInit = true; // Force first init
+                if (value) _ = Init();
+                else Dispose(); // Dispose instance if user disabled database function globally
+            }
+        }
+        
+        
+        private static string _uri;
+        public static string Uri
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(_uri)) return _uri;
+                var c = DbConfig.DbUrl;
+                _uri = c;
+                return c;
+            }
+            set
+            {
+                if (value != _uri) _isFirstInit = true; // Force first init if value changed
+                
+                _uri           = value;
+                DbConfig.DbUrl = value;
+                _isFirstInit   = true;
+            }
+        }
+
+        private static string _token;
+        public static string Token
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(_token)) return _token;
+                var c = DbConfig.DbToken;
+                _token = c;
+                return c;
+            }
+            set
+            {
+                if (value != _token) _isFirstInit = true; // Force first init if value changed
+                
+                _token           = value;
+                DbConfig.DbToken = value;
+                _isFirstInit     = true;
+            }
+        }
+
+        private static Guid?  _userId;
+        private static string _userIdHash;
+        public static Guid UserId
+        {
+            get
+            {
+                if (_userId != null) return (Guid)_userId;
+                var c = DbConfig.UserGuid; // Get or create (if not yet has one) GUIDv7
+                _userId = c;
+                _userIdHash = BitConverter.ToString(System.IO.Hashing.XxHash64.Hash(c.ToByteArray())).Replace("-", "")
+                                          .ToLowerInvariant(); // Get hash for the GUID to be used as SQL table name
+                // I know that this is overkill, but I want it to be totally non-identifiable if for some reason someone
+                // has access to their database. It also lowers the amount of query command length to be sent, hopefully
+                // reducing access latency.
+                // p.s. oh yeah, this is also why user won't be able to get their data back if they lost the GUID,
+                // good luck reversing Xxhash64 back to GUIDv7. Technically possible, but good luck!
+                return c;
```

Comment:
Since the table name is hashed, is it necessary to have the user input the user id with uuid? You can add salt if you feel it's easily get reversed.

Reply:
I was thinking it on how easy it is to generate the GUID on hand and parse it to see if its a valid one without compromising the security of the database if the table names were dumped (hashed guid can't directly related to a user, unlike a direct user-inputted user ID). The reverse-ability in the comment is purely a joke.

Reply:
> I was thinking it on how easy it is to generate the GUID on hand and parse it to see if its a valid one without compromising the security of the database if the table names were dumped (hashed guid can't directly related to a user, unlike a direct user-inputted user ID). The reverse-ability in the comment is purely a joke.

Since the user id is hashed, it's not recognizable for hackers. It's also more userfriendly cuz we don't need to sync the GUID in another way.

Reply:
Got it, I'll try to move to random word generator and see how it does 

Ref https://stackoverflow.com/questions/18110243/random-word-generator-2

Reply:
I think we can just let user input anything they want, and we hash it with salt like "collaspe-". Also sqlite does not have permission system, so one should use his own server. Do we really need a user id?

Reply:
UserID still needed for the instance where a user wants to have a shared database server but doesn't share the data

Reply:
So now the security is based on the RANDOM UUID to get rid of being overrided by others with same user id. :(

Reply:
I'm not sure what you meant, my brain is melting due to heat today. 
But we have a few choices for this one
1. Keep the logic as is, where the UserID uses UUIDv7 that is hashed. This is simpler logically and more secure, but not very user friendly as UUIDv7 is pretty long.
2. Move from UUIDv7 to random words (still editable by users) that is hashed. Should be more user friendly but maybe less secure since user might just put a very simple UserID.

Reply:
I'm not sure what you meant, my brain is melting due to heat today. 
But we have a few choices for this one
1. Keep the logic as is, where the UserID uses UUIDv7 that is hashed. This is simpler logically and more secure, but not very user friendly as UUIDv7 is pretty long.
2. Move from UUIDv7 to random words (still editable by users) that is hashed. Should be more user friendly but maybe less secure since user might just put a very simple UserID. Tbh this still sufficient since its the user choice to share their database so, eh.

Reply:
generate UUID, but editable by users and not limited to UUID format.

Reply:
Adjusted in https://github.com/CollapseLauncher/Collapse/pull/586/commits/2dede52ab45a50354713637ba728e9e62f77c8df